_reports/2023/goegge.md

```diff
@@ -0,0 +1,310 @@
+were being run.  If new tests were needed, Niklas often contributed
+them.  Reviewers frequently thanked Niklas publicly and several
+developers we spoke to privately indicated that positive reviews from
+Niklas were a strong signal that a pull request was high quality and had
+a good approach, encouraging other developers to contribute reviews and
+accelerating the progress of those pull requests towards getting merged.
+
+In at least four pull requests ([#28948][], [#28685][], [#28578][], and
+[#29242][]), Niklas used tests to find bugs before they were merged.  In
+particular, #28685 is a fix for what could have become a serious bug
+that was caught by fellow Brink engineer Fabian Jahr (with diagnosis of
+the underlying issue by fellow Brink engineer Sebastian Falbesoner).
+After discovery of the bug, Niklas wrote a fuzz testing harness that
+found two additional serious bugs and allowed them to be fixed at the
+same time.
+
+Press:
+
+- Bitcoin UTXO set summary hash replacement - **Bitcoin Optech Newsletter #274** [¶](https://bitcoinops.org/en/newsletters/2023/10/25/#bitcoin-utxo-set-summary-hash-replacement)
+- Bitcoin Core #28685 fixes a bug in the calculation of the hash of a UTXO set - **Bitcoin Optech Newsletter #275** [¶](https://bitcoinops.org/en/newsletters/2023/11/01/#bitcoin-core-28685)
+
+## Fuzz testing net processing
+
+> In my opinion, improving our testing is the most important and
+> impactful thing to work on. It would reduce our review bottleneck, it
+> would make our code easier to change, and it would cause less
+> frustration among developers.
+
+Fuzz testing takes code that was designed in the head of a programmer
+who probably expected an ideal situation and exposes it to variety of
+semi-randomly-generated input to see what happens.  In a poorly designed
+program, exposing it to unexpected input can lead to crashes,
+memory leaks, wrong states, disclosure of private state, and many other
+problems.  In mission-critical software like Bitcoin Core, any of those
+problems can quickly become a major vulnerability: a crash of a full
+node can allow theft of funds from a related Lightning Network wallet; a
+memory leak can escalate to a crash; wrong states can lead to accepting
+invalid transactions; and disclosure of private state can lead to theft.
+
+Before fuzz testing was used with Bitcoin Core, developers were already
+on the cutting edge of using defensive programming techniques to avoid
+the worst problems, but [many problems][topic cve-2018-17144] have still
+found their way into the code (and more probably exist).  Fuzz testing
+allows developers to leverage cheap and abundant CPU and memory to test
+functions with massive numbers of random inputs to see if something
+fails.
+
+Niklas has successfully used fuzz testing to discover multiple bugs that
+could have led to the loss of bitcoins if an attacker had discovered
+them first.  Most Bitcoin Core developers are supportive of increased
```

Comment:
Should maybe lead with this. Bug hunter Niklas is slaying the bugs before they come steal your bitcoins!

Reply:
Moved this paragraph up